### ISSUE TITLE
Debian 7.8 build error no soup cache?

### ISSUE THREAD

**Issue by soyuka** [fanglingsu/vimb#187]
Hi,

Wanted to test this on my debian PC but I get stuck on :

```
main.h:350:5: error: unknown type name ‘SoupCache’
```

I checked, `libsoup2.4-dev` is installed.

**Comment by fanglingsu** [comment]
Thank you for the issue.  I'm not sure what's to problem here according to https://developer.gnome.org/libsoup/stable/SoupCache.html the SoupCache exists since souop version 2.34. Maybe this is caused by missed include in main.h.

Could you please check if compilation works if `#include <libsoup/soup.h>` is added to the includes in main.h?

**Comment by soyuka** [comment]
Thanks for the quick answer.

That's the first thing I did but had the same error. 

My libsoup is located under `/usr/include/libsoup-2.4/libsoup/`, so it should work.

**Comment by iamleot** [comment]
Hello Antoine, Daniel and the entire vimb community,
Antoine Bluchet writes:
> Thanks for the quick answer.
> 
> That's the first thing I did but had the same error. 
> 
> My libsoup is located under `/usr/include/libsoup-2.4/libsoup/`, so it should work.
Maybe the output of the following commands will be helpful too:

 $ pkg-config --libs libsoup-2.4
 $ pkg-config --cflags libsoup-2.4


HTH,
Ciao,
L.

**Comment by fanglingsu** [comment]
Hi Antoine,

this is the output that Leonardo mentioned. Maybe this helps you to inverstigate into the issue.

    $ pkg-config --libs libsoup-2.4
    -lsoup-2.4 -lgio-2.0 -lgobject-2.0 -lglib-2.0

    $ pkg-config --cflags libsoup-2.4
    -pthread -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include
    
Some other things:

1. Have a look at /usr/include/libsoup-2.4/libsoup/soup-version.h SOUP_MAJOR_VERSION and SOUP_MINOR_VERSION numbers.
2. Does the /usr/include/libsoup-2.4/libsoup/soup-cache.h exists.

**Comment by soyuka** [comment]
```
❯ pkg-config --libs libsoup-2.4
-lsoup-2.4 -lgio-2.0 -lgobject-2.0 -lglib-2.0
❯ pkg-config --cflags libsoup-2.4
-pthread -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include
❯ [ -f /usr/include/libsoup-2.4/libsoup/soup-cache.h ] && echo ok
ok
```

Everything seems alright, any of you successfuly installed vimb on debian? I can't find the `soup-version` file thought.

**Comment by fanglingsu** [comment]
OK the soup-version.h was added in 2.42 so your libsoup seems to be a little older.

    $ grep '#define SOUP_CACHE(' /usr/include/libsoup-2.4/libsoup/soup-cache.h
    define SOUP_CACHE(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), SOUP_TYPE_CACHE, SoupCache))

**Comment by soyuka** [comment]
What I did :

1. libsoup needs glib >= 2.36.0, wheezy has 2.32.4 compiled [2.42 from the repo](https://git.gnome.org/browse/glib/?h=glib-2-42)
2. Compile libsoup 2-42 ([master](https://github.com/djdeath/libsoup)) with `--disable-tls-check` (was lazy to compile glib-networking)

 Don't forget to `ldconfig`.


And tadam, I got vimb working :).

Thanks guys for your leads! Hope that this'll improve in the future (maybe with a package).
